### ISSUE TITLE
[Cost Insights] Currencies

### ISSUE THREAD

**Issue by vannarath-poeu** [backstage/backstage#6695]
Is it possible for us to override the defaultCurrencies in config? Also, is it possible to set the defaultCost to USD instead of engineers>

## Feature Suggestion

I am raising this request as there are organisations that spread across regions and the engineering cost is not applicable as they vary between teams.

## Possible Implementation

One possible implementation is to allow override the defaultCurrencies in the config similar to products.
Also, we can update the CurrencyProvider to read a default currency in config if it exists with a fallback to engineers.

**Comment by timbonicus** [comment]
It's not possible in configuration today, but this definitely makes sense. The currencies were intended to be extensible, it just isn't implemented as such yet. The [defaultCurrencies](https://github.com/backstage/backstage/blob/master/plugins/cost-insights/src/utils/currency.ts#L32) show the currency fields (`engineers` is treated a bit special as it's time-based and a bit more complex than simple conversion).

This could also be code-configured in `App.tsx` similar to how you can pass in [catalog table columns](https://github.com/backstage/backstage/blob/4f3c36bf42205f5a9181437630721f980d68105a/plugins/catalog/src/components/CatalogPage/CatalogPage.tsx#L53).

**Comment by vannarath-poeu** [comment]
Hi @timbonicus,

Thanks for the prompt response and advice (especially the second one). I would appreciate it if you could have a look to see if my pull request would help this.

The intention is to let users pass in an optional prop to replace the defaultCurrencies in useConfig. Then, in the useCurrency, we will be reading from the config and if engineers exist in the list, it will be the default, else the first one will be. This allows the users to overwrite the currencies and still maintain backward compatibility to use defaultCurrencies should the users choose to not pass anything in.

**Comment by ryanvazquez** [comment]
Hey @vannarath-poeu, thanks for the PR! Def reasonable to have currencies be configurable (we had actually planned for this a while back except the expected implementation was via `app.info-yaml` which `ConfigProvider` reads from).

I think this request brushes up against a larger question of how plugin configuration/customization should work moving forward, which I think warrants some discussion. Via the suggested approach, cost insights would be customizable in several ways - via client (such as alerts), `app.info-yaml`, and now the exported Page component.

I think ideally users should have a single, simple interface to customize the plugin to their needs and adding an additional surface would complicate setup and create too many sources-of-truth for the plugin which can lead to unnecessary downstream complexity. 

I think we should  put your PR on the backburner for now and open up an issue where we might discuss further.

**Comment by vannarath-poeu** [comment]
Hi @ryanvazquez / @timbonicus,

Thank you so much for helping with this. These are really great insights. I have closed my previous pull request for now.

@ryanvazquez Thank you for initiating the RFC. I believe that would take quite some time and it should. Can I suggest a 2-pronged strategy to tackle this?
I have opened another pull request https://github.com/backstage/backstage/pull/6717 to allow passing in an optional currencies config. This should just extend the functionality and not add any confusion. Let me know what you think.
